File: Triatlon-C#/Triatlon/repository/interfaces/ParticipantRepository.cs
using System;
using System.Collections.Generic;
using ConcursTriatlon.domain;

namespace ConcursTriatlon.repository.interfaces
{
    public interface ParticipantRepository : Repository<long, Participant>
    {
        List<long> findParticipantByName(String nume);

        Dictionary<Participant, long> GetParticipantiCuPunctaj();
    }
}

File: Triatlon-Java/TriatlonPersistance/src/main/java/repository/interfaces/ParticipantRepository.java
package repository.interfaces;

import domain.*;
import repository.Repository;

import java.util.List;
import java.util.Map;

public interface ParticipantRepository extends Repository<Long, Participant> {

    List<Participant>   findParticipantByName(String nume);

    public Map<Participant, Long> getParticipantiCuPunctaj();

}
